doc/nrf/app_approtect.rst

```diff
@@ -0,0 +1,110 @@
+**********************
+
+The access port protection mechanism is configured in two steps: in the hardware and in the firmware.
+
+Hardware step
+=============
+
+The hardware step consists of the verification of the mechanism state by fetching the value of ``UICR.APPROTECT``.
+This takes place at the device reset, and the mechanism state is different for different SoCs:
+
+* SoCs of the nRF52 and nRF53 Series: Default ``UICR.APPROTECT`` value prevents debugger access (APPROTECT is enabled)
+* nRF9160 SoC: Default ``UICR.APPROTECT`` value does not prevent debugger access (APPROTECT is disabled)
+
+Software step
+=============
+
+The software step takes place during startup, with the execution of a function provided by the ``hal_nordic`` module.
+This function writes a piece of debug code to the Access Port's ``APPROTECT.DISABLE`` register, which lets you debug the firmware.
+This step is valid for SoCs of the nRF52 and nRF53 Series, although the value length will differ depending on the SoC type:
+
+* SoCs of the nRF52 Series: ``0x5A`` (8-bit value)
+* SoCs of the nRF53 Series: ``0x5A0F`` (32-bit value)
+
+This step can be automated using nrfjprog.
+
+.. _app_approtect_ncs:
+
+Access port protection configuration options in the |NCS|
+*********************************************************
+
+Depending on the SoC you are using, you can use the following Kconfig options to enable access port protection mechanism in the |NCS|:
+
+* SoCs of the nRF52 and nRF53 Series: :kconfig:option:`CONFIG_NRF_APPROTECT_LOCK`
+* SoCs of the nRF52 and nRF53 Series, and nRF9160: :kconfig:option:`CONFIG_NRF_APPROTECT_USE_UICR`
+
+.. _app_approtect_ncs_lock:
+
+Enabling access port protection mechanism with :kconfig:option:`CONFIG_NRF_APPROTECT_LOCK`
+==========================================================================================
+
+Setting the :kconfig:option:`CONFIG_NRF_APPROTECT_LOCK` Kconfig option to ``y`` and compiling the firmware is enough to enable the access port protection mechanism for SoCs of the nRF52 and nRF53 Series.
+The access port protection configured in this way cannot be disabled without erasing the flash.
+
+.. _app_approtect_ncs_use_uicr:
+
+Enabling access port protection mechanism with :kconfig:option:`CONFIG_NRF_APPROTECT_USE_UICR`
+==============================================================================================
+
+Setting the :kconfig:option:`CONFIG_NRF_APPROTECT_USE_UICR` Kconfig option to ``y`` and compiling the firmware creates your application in two variants: one that blocks any debugger access and one that lets you access for the debugger.
+
+You can start debugging the firmware variant that allows it without additional steps needed.
+
+Once you are done debugging, complete the following steps to enable the access port protection on this variant:
+
+1. Write to the register the value that blocks debugger access:
+
+   .. code-block:: console
+
+      nrfjprog --memwr 0x10001208 --val 0x00
```

Comment:
Use nrfjprog --rbp all here, as addresses and values varies across device families.